### PR TITLE
OCPBUGS-7116: Console operator should sync managed cluster 'controlPlaneTopology' field to managed cluster config

### PR DESCRIPTION
**Pull request by jhadvig**
Implemented per our discussion. The `controlPlaneTopology` will be part of the manages cluster config ConfigMap.

/assign @TheRealJon

### CODE REVIEW

File: pkg/console/controllers/advancedclustermanagement/managedclusters/controller.go [hunk 1]
```diff
@@ -173,6 +174,10 @@ func (c *ManagedClusterController) Sync(ctx context.Context, controllerContext f
 	errReason, err = c.SyncOLMConfigManagedClusterViews(ctx, operatorConfig, managedClusters)
 	statusHandler.AddConditions(status.HandleProgressingOrDegraded("ManagedClusterSync", errReason, err))
 
+	// Create managed cluster views for Infrastructure config
+	errReason, err = c.SyncOLMConfigManagedClusterViews(ctx, operatorConfig, managedClusters)
```
Comment (TheRealJon) [inline pkg/console/controllers/advancedclustermanagement/managedclusters/controller.go:178]:
```suggestion
	errReason, err = c.SyncInfrastructureConfigManagedClusterViews(ctx, operatorConfig, managedClusters)
```

File: pkg/console/controllers/advancedclustermanagement/managedclusters/controller.go [hunk 2]
```diff
@@ -517,6 +547,12 @@ func (c *ManagedClusterController) SyncManagedClusterConfigMap(managedClusters [
 			continue
 		}
 
+		controlPlaneTopology, err := managedclusterviewsub.GetInfrastructureConfigCopiedCSVDisabled(olmConfigMCV)
+		if err != nil {
+			klog.V(4).Infof("Error getting copiedCSVDisabled field for managed cluster %v", clusterName)
+			continue
+		}
+
```
Comment (TheRealJon) [inline pkg/console/controllers/advancedclustermanagement/managedclusters/controller.go:555]: Don't we need to fetch the infrastructure config MCV as we did for OLMConfig?
```suggestion
		// Check that managed cluster infrastructure config MCV has already been synced, skip if not found
		infraConfigMCV, err := c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(clusterName).Get(ctx, api.InfrastructureConfigManagedClusterViewName, metav1.GetOptions{})
		if apierrors.IsNotFound(err) {
			klog.V(4).Infof("Infrastructure config ManagedClusterView not found for managed cluster %v", clusterName)
			continue
		}
		// Skip if unable to get olm config MCV for any other reason
		if err != nil {
			klog.V(4).Infof("Error getting Infrastructure config ManagedClusterView for managed cluster %v", clusterName)
			continue
		}
		controlPlaneTopology, err := managedclusterviewsub.GetInfrastructureConfigControlPlaneTopology(infraConfigMCV)
    if err != nil {
			klog.V(4).Infof("Error getting controlPlaneTopology field for managed cluster %v", clusterName)
			continue
		}
```

File: pkg/console/subresource/managedclusterview/infrastructure_config.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	}
+	return mcv, nil
+}
+
+func InfrastructureConfigViewStub(cn string) *unstructured.Unstructured {
+	return util.ReadUnstructuredOrDie(assets.MustAsset("managedclusterviews/infrastructure-config.yaml"))
+}
+
+func GetInfrastructureConfigCopiedCSVDisabled(olmConfig *unstructured.Unstructured) (string, error) {
+	controlPlaneTopology, found, err := unstructured.NestedString(olmConfig.Object, "status", "result", "status", "controlPlaneTopology")
```
Comment (TheRealJon) [inline pkg/console/subresource/managedclusterview/infrastructure_config.go:34]:
```suggestion
func GetInfrastructureConfigControlPlaneTopology(infraConfig *unstructured.Unstructured) (string, error) {
	controlPlaneTopology, found, err := unstructured.NestedString(infraConfig.Object, "status", "result", "status", "controlPlaneTopology")
```

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Missed a couple of name changes

**Comment by jhadvig** [comment]
@TheRealJon comments addressed.
Please check last commit, also added some minor refactoring.

**Comment by jhadvig** [comment]
Closing since we are going to remove the ACM related code.

cc'ing @TheRealJon

**Pull request closed without merging**